### DISCUSSION TITLE
How to use groupLimit with camelCase fields in postgress?

### DISCUSSION THREAD

**Discussion by yariksav** [adonisjs/core#2047] [Help]
Postgres has a behavior to put fields name of SQL query into lowerCase, and then if I have field "firmId", it throws error:

select * from (select *, row_number() over (PARTITION BY firmId ORDER BY id desc) as adonis_group_limit_counter from "organization"."objects" where "firmId" in ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13)) as "adonis_temp" where "adonis_group_limit_counter" <= $14 - column "firmid" does not exist

Maybe needs to cover field into quotes. But I am not sure
https://github.com/adonisjs/lucid/blob/f189e57861fd1599b65bde167c7717f4de876d03/src/Orm/Relations/HasMany/QueryBuilder.ts#L134
```
const partitionBy = `PARTITION BY "${this.relation.foreignKeyColumName}"`
```

Thanks

**Comment by thetutlage** [comment]
What's the code you are writing that produces this query?

**Comment by yariksav** [comment]
Hi, @thetutlage, thank you for answer.

My database is camelCase fields named, then
For example i have models
```
export default class Objects extends BaseModel {
  public static table = 'organization.objects'

  @column({ isPrimary: true })
  public id: string

  @column()
  public firmId: string
  ...
}
```

```
export default class Firm extends BaseModel {
  public static table = 'organization.firms'

  @column({ isPrimary: true })
  public id: string

  @hasMany(() => Objects, {
    foreignKey: 'firmId'
  })
  public objects: HasMany<typeof Objects>
}
```

Query is 
```
  const res = await Firm.query()
      .preload('objects', q => q.groupLimit(2)) // 👈
```


*> I disabled auto conversion fileds by framework to underscore in database.ts for be able to use camelCase fileds
```
 orm: {
    getColumnName (_, key: string) {
      return key
    },
    getSerializeAsKey (_, key: string) {
      return key
    }
  }
```

Thanks

**Reply by thetutlage** [reply]
Thanks for sharing the details. Yup looks like the columnName inside partition needs to be inside quotes too. I will look into it and publish a fix

**Reply by yariksav** [reply]
Thank you!!!

**Comment by yariksav** [comment]
Yes, I tried every possibility to fix it by column names, for example **foreignKey: '"firmId"'**  but is brakes whole requests.
then I tried to change code locally in node_modules

https://github.com/adonisjs/lucid/blob/f189e57861fd1599b65bde167c7717f4de876d03/src/Orm/Relations/HasMany/QueryBuilder.ts#L134

```
        const partitionBy = `PARTITION BY "${this.relation.foreignKeyColumName}"`;
        const orderBy = `ORDER BY "${column}" ${direction}`;
```

(i'm not sure for second row ORDER BY "${column}", if someone will need complex row with schema name... groupOrderBy('blog.postId') -> it can be done by groupOrderBy('blog."postId"'), but without first row I cant do grouping
